### DISCUSSION TITLE
How to get same results for each simulation run?

### DISCUSSION THREAD

**Discussion by yossefyossef** [NVlabs/sionna#82] [Q&A] [answered]
Hi all,
Maybe someone can help me please?

I run the LDPC code in the following link:
https://nvlabs.github.io/sionna/examples/5G_Channel_Coding_Polar_vs_LDPC_Codes.html
for each run, I get different BLER result up to 6% difference.

I would like to get the same results for all runs.

I added the following rows, but it does not help.

import random
random.seed(1, 2)

Can someone please help me?

Thanking in advance, and best regards,
Yossef Yossef
Email: yossiy@gilat.com

**Comment by jhoydis** [comment, marked as answer]
Hi Yossef,

What you observe is very likely due to an insufficient number of Monte Carlo trials.
Have you tried to increase the values of ``max_mc_iter`` and ``num_target_block_errors``?

The other thing to keep in mind is that the random number generation seed has no effect in XLA mode, see https://www.tensorflow.org/xla/known_issues#random_number_generation_ignores_tf_seed.
If you run the simulations in standard graph mode and set a seed, you should get the same results for each run.

In your code example, you are setting the seed of the Python random number generation, not the one of TensorFlow. The correct function to use is https://www.tensorflow.org/api_docs/python/tf/random/set_seed.

Hope this helps.

**Comment by nbecker** [comment]
On Mon, Dec 19, 2022 at 3:29 AM yossefyossef ***@***.***>
wrote:

> Hi all,
> Maybe someone can help me please?
>
> I run the LDPC code in the following link:
>
> https://nvlabs.github.io/sionna/examples/5G_Channel_Coding_Polar_vs_LDPC_Codes.html
> for each run, I get different BLER result up to 6% difference.
>
> I would like to get the same results for all runs.
>
> I added the following rows, but it does not help.
>
> import random
> random.seed(1, 2)
>
> Can someone please help me?
>
> Thanking in advance, and best regards,
> Yossef Yossef
> Email: ***@***.***
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NVlabs/sionna/discussions/82>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA4VUFFF4AKO4QJU3M6UN3WOAMHDANCNFSM6AAAAAATDCXHZU>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
maybe you need: tf.random.set_seed(1)

-- 
*Those who don't understand recursion are doomed to repeat it*

**Reply by yossefyossef** [reply]
Yes, it seems that it helps, I am working on it. HUGE THANKs to all of you!
